File: Daily Test/C/Find Word By Middle Letter 14.9.2025.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char S[101];
    scanf("%s", S);
    int N;
    scanf("%d", &N);
    char arr[N][101];
    for (int i = 0; i < N; i++)
        scanf("%s", arr[i]);

    for (int si = 0; si < strlen(S); si++) {
        char c = S[si];
        int found = 0;
        for (int j = 0; j < N; j++) {
            int mid = strlen(arr[j]) / 2;
            if (tolower(arr[j][mid]) == tolower(c)) {
                printf("%s\n", arr[j]);
                found = 1;
                break;
            }
        }
        if (!found) {
            int mid = strlen(arr[0]) / 2;
            int len = strlen(arr[0]);
            for (int k = 0; k < mid; k++) printf("*");
            printf("%c", c);
            for (int k = mid + 1; k < len; k++) printf("*");
            printf("\n");
        }
    }
    return 0;
}

File: Daily Test/Java/Find Word By Middle Letter 14.9.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String S = sc.nextLine().trim();
        int N = sc.nextInt();
        sc.nextLine();
        String[] arr = new String[N];
        for (int i = 0; i < N; i++) arr[i] = sc.nextLine().trim();

        for (char ch : S.toCharArray()) {
            boolean found = false;
            for (String j : arr) {
                int mid = j.length() / 2;
                if (Character.toLowerCase(j.charAt(mid)) == Character.toLowerCase(ch)) {
                    System.out.println(j);
                    found = true;
                    break;
                }
            }
            if (!found) {
                if (arr.length > 0) {
                    int mid = arr[0].length() / 2;
                    for (int k = 0; k < mid; k++) System.out.print("*");
                    System.out.print(ch);
                    for (int k = mid + 1; k < arr[0].length(); k++) System.out.print("*");
                    System.out.println();
                } else {
                    System.out.println(ch);
                }
            }
        }
    }
}